src/main/java/org/jboss/draw/Lottery.java

```diff
@@ -0,0 +1,98 @@
+package org.jboss.draw;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.mailer.Mail;
+import io.quarkus.mailer.Mailer;
+import io.smallrye.mutiny.tuples.Tuple2;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.jboss.config.LotteryConfig;
+import org.jboss.config.LotteryConfigProducer;
+import org.jboss.draw.entities.Issue;
+import org.jboss.draw.entities.Participant;
+import org.jboss.processing.state.EveryIssueState;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Unremovable
+@ApplicationScoped
+public class Lottery {
+
+    @Inject
+    private LotteryConfigProducer lotteryConfigProducer;
+
+    @Inject
+    private Mailer mailer;
+
+    private static final Pattern USERNAME_FROM_EMAIL = Pattern.compile("\\s+(\\w+)@");
```

Comment:
there must be a space in email?

Reply:
That is just in case to skip white chars in the beginning, if there is a typo

Reply:
still needs to be redone